osquery/logger/CMakeLists.txt

```diff
@@ -18,6 +18,17 @@ else()
   ADD_OSQUERY_LINK_ADDITIONAL("aws-cpp-sdk-core")
 endif()
 
+if(NOT WINDOWS)
+  ADD_OSQUERY_LINK_ADDITIONAL("z")
```

Comment:
We should bubble up these general links to the cmake lists in the directory above. If these were giving you issues it may be a problem with the order of links. We'll need to make sure the general dependencies are included at the end.